docs/report_stage_iii.md

```diff
@@ -309,144 +334,257 @@ Calculating the averages, we get the following alternative gas cost schedule:
+SWAP9|3|2.01|-32.99%|16.21%
+SWAP10|3|2.37|-21.06%|19.15%
+SWAP11|3|1.99|-33.59%|15.01%
+SWAP12|3|2.12|-29.19%|15.98%
+SWAP13|3|2.13|-29.03%|16.78%
+SWAP14|3|1.97|-34.46%|15.40%
+SWAP15|3|2.38|-20.72%|21.62%
+SWAP16|3|2.29|-23.70%|19.30%
+
+
+> **Remarks**
 >
 > The data clearly shows that some changes can be made to the current gas proposal. Still, any such changes should be carefully considered, as there are other factors not taken into account in this research: ease of implementation, backward compatibility, hard fork requirement, existing tooling, hardcoded values in auxiliary software, etc.
 >
 >The alternative gas cost schedule is a proposal and should be treated as such. We believe that the methodology presented in this report is a good starting point for further research and discussion.
+
+
+### EXP
+For the detailed analysis of the `EXP` opcode, we have used the following data:
+- [EvmOne](./report_stage_iii_assets/evmone_measure_arguments_single_exp.html)
+- [Geth](./report_stage_iii_assets/geth_measure_arguments_single_exp.html)
+- [Netehrmind](./report_stage_iii_assets/nethermind_measure_arguments_single_exp.html)
+- [EtherumJS](./report_stage_iii_assets/ethereumjs_measure_arguments_single_exp.html)
+- [Erigon](./report_stage_iii_assets/erigon_measure_arguments_single_exp.html)
+- [Besu](./report_stage_iii_assets/besu_measure_arguments_single_exp.html)
+- [Rust EVM](./report_stage_iii_assets/revm_measure_arguments_single_exp.html)
+
+Please see 'EXP' section in the corresponding files.
+
+The `EXP` opcode proved quite difficult to estimate due to high client variability. Additionally, the measure marginal method was not able to sufficiently estimate the base cost of the opcode. The `EXP` measure marginal method used the argument size of 1 byte, as this provided the most consistent results. Still, for some clients, the measured cost was below 0.
+
+For a more accurate estimation, we compare EXP base and dynamic times to the execution of `ADD` and `MUL` opcodes. These opcodes are used as a reference as they are the most basic arithmetic ones. Given this data we can estimate the cost of the single byte exponentiation as follows:
+
+```math
+	exp_byte_cost = exp_byte_time / (add_time + mul_time) * (add_cost + mul_cost)
+```
+
+With this formula we can further estimate the base cost of the `EXP` opcode, by subtracting the cost of the arguments from the total cost of `EXP`. We disregard values below 0.
+
+The following table shows the results:
+
+Client|ADD ns|MUL ns|EXP per byte ns|EXP per byte gas| EXP base ns | EXP base gas
+:----- | ----: | -----: | ----: | -----: | -----: | -----:
+EvmOne|18.30|30.34|214.25|35.23|15.35|0
+Geth|19.78|28.21|98.74|16.46|82.88|0
+Netehrmind|31.44|61.78|274.48|23.55|870.21|51.12
+EtherumJS|290|381|1590|18.96|4623|36.16
+Erigon|6.68|7.81|84.63|46.72|0|0
+Besu*|58815|75348|154635|9.22|415094|15.53
+Rust EVM|2.86|3.59|35.13|43.57|15.38|0
+
+*Besu results inflated by the internal loop of 10k times.
```

Comment:
Even more. Recall that results should not be compared between EVMs due to various benchmark methods used.